app/contribution-hub/components/constraintAppDetailModal.tsx

```diff
@@ -0,0 +1,224 @@
+
+  const [isCollectionValid, setIsCollectionValid] = useState<boolean>(false);
+
+  const [decimals, setDecimals] = useState<number | undefined>();
+
+  const [constraint, setConstraint] = useState<ConstraintProps | null>(null);
+
+  const createRequirementParamsList = useCallback(() => {
+    if (!constraint) return;
+
+    if (constraint.params.length > 0) {
+      setRequirementParamsList(
+        constraint.params.reduce((obj: any, item: any, index: any) => {
+          obj[item] = "";
+          return obj;
+        }, {}),
+      );
+    }
+  }, [constraint]);
+
+  const [errorMessage, setErrorMessage] = useState<string | null>();
+
+  useEffect(() => {
+    const requirement = requirementList.find(
+      (item) => item.pk == constraint?.pk,
+    );
+
+    setExistRequirement(requirement ?? null);
+
+    if (requirement) {
+      setIsNotSatisfy(requirement.isNotSatisfy);
+      setRequirementParamsList(requirement.params);
+    } else {
+      createRequirementParamsList();
+    }
+  }, [constraint, createRequirementParamsList, requirementList]);
+
+  const checkingParamsValidation = () => {
+    if (!requirementParamsList) return false;
+    if (
+      !requirementParamsList.ADDRESS ||
+      !requirementParamsList.CHAIN ||
+      !requirementParamsList.MINIMUM ||
+      Number(requirementParamsList.MINIMUM) <= 0
+    ) {
+      !requirementParamsList.ADDRESS
+        ? setErrorMessage("Please enter collection address.")
+        : !requirementParamsList.CHAIN
+          ? setErrorMessage("Please select chain.")
+          : setErrorMessage("Please select minimum amount.");
+      return false;
+    }
+
+    if (!isCollectionValid) return false;
+    return true;
+  };
+
+  const checkCsvFileUploadedValidation = () => {
+    if (!requirementParamsList) return false;
+    if (!requirementParamsList.CSV_FILE) {
+      setErrorMessage("Please upload a csv file.");
+      return false;
+    }
+    return true;
+  };
+
+  const handleAddRequirement = () => {
+    if (!constraint) return;
+
+    if (
+      constraint.name === "core.HasNFTVerification" ||
+      constraint.name === "core.HasTokenVerification"
+    ) {
+      const res = checkingParamsValidation();
+      if (!res) return;
+    }
+
+    if (constraint.name === "core.AllowListVerification") {
+      const res = checkCsvFileUploadedValidation();
+      if (!res) return;
+    }
+
+    addRequirements(
+      existRequirement,
+      constraint.pk,
+      constraint.name,
+      constraint.title,
+      isNotSatisfy,
+      requirementParamsList,
+      constraintFile,
+      decimals,
+    );
+
+    setSelectedApp();
+  };
+
+  const handleSelectNotSatisfy = (isSatisfy: boolean) => {
+    setIsNotSatisfy(isSatisfy);
+  };
+
+  return (
+    <div className="min-h[200px] mt-5 flex flex-col gap-2">
+      <div
+        className="absolute top-5 z-[999] cursor-pointer"
+        onClick={() => {
+          setSelectedApp();
+        }}
```

Comment:
pass the function instead of creating another function

Reply:
it should be like that because the argument should be passed as undefined instead of the onclick event